test/unit/validators/non_localhost_validator_test.rb

```diff
@@ -14,33 +14,33 @@ def test_validate_each
     assert_nil errors
     assert record.errors.empty?
 
-    errors = validator.validate_each(record, :api_backend, 'http://localhost/')
-    assert_equal ["can't be localhost"], errors
-    assert record.errors.present?
+    error = validator.validate_each(record, :api_backend, 'http://localhost/')
+    assert_equal ActiveModel::Error, error.class
+    assert "can't be localhost", record.errors.messages_for(:api_backend)
```

Comment:
`"can't be localhost"` will always be true. Did you mean to pass arguments the other way around? Also, to make sure messages are not an empty array, perhaps still assert `record.errors.messages_for(:api_backend).present?`

Reply:
Yes, of course I meant to have `assert_equal` there instead of `assert` :upside_down_face:  Good catch!